packages/core/src/controllers/relayer.ts

```diff
@@ -334,6 +334,23 @@ export class Relayer extends IRelayer {
     throw new Error("No internet connection detected. Please restart your network and try again.");
   }
 
+  public async handleBatchMessageEvents(messages: RelayerTypes.MessageEvent[]) {
+    if (messages?.length === 0) {
+      this.logger.trace("Batch message events is empty. Ignoring...");
+      return;
+    }
+    try {
+      const sortedMessages = messages.sort((a, b) => a.publishedAt - b.publishedAt);
+      this.logger.trace(`Batch message events sorted`);
```

Comment:
Would be good to trace the batch size too:

```suggestion
      this.logger.trace(`Batch of ${sortedMessages.length} message events sorted`);
```